### PR TITLE
Enable VR update for SP7 platforms

### PR DESCRIPTION
**Pull request by abinayaddhandapani**
1. SP7: Move CXX standard to 23 and fix compilation issues.
2. SP7: Add support for Congo platform.
3. SP7: Add support for Morocco platform

### CODE REVIEW

**Comment by ojayanth** [comment]
Please add summary in PR summary about the scope of this PR.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.8.10)
 project(vr-update CXX)
 
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 23)
```
Comment (ojayanth) [inline CMakeLists.txt:4]: Commit message:    SP7: build: upgrade to C++23

File: src/vr_update_mps.cpp [hunk 2]
```diff
@@ -29,7 +29,7 @@ bool vr_update_mps::crcCheckSum(){
      }
 
     ret = i2c_smbus_read_i2c_block_data(fd, CRC_ADDR, BYTE_COUNT_4, rdata);
-    if (rdata < SUCCESS)
+    if (ret < SUCCESS)
```
Comment (ojayanth) [inline src/vr_update_mps.cpp:32]: is this bug? same for others also , if it is bug use diffrent commit to fix this.

File: inc/vr_update.hpp [hunk 3]
```diff
@@ -114,6 +114,10 @@ extern "C"
 #define SLEEP_1               (1000000)
 #define SLEEP_2               (2000000)
 #define SLEEP_1000            (1000)
+#define INDEX_40              (0x40)
```
Comment (ojayanth) [inline inc/vr_update.hpp:117]: Is this code improvement or  c+23 is not happy with multiple places defeinition?

**Review by ojayanth — COMMENTED** [review]
Not a domain experts , Reviwed only coding guidlnes perspective

File: src/vr_update_mps.cpp (hunk 2 above)
Comment (abinayaddhandapani) [reply src/vr_update_mps.cpp:32]: These are warnings which are not reported as part of C++17.
But when migrated to C++23 , this is throwing error

File: inc/vr_update.hpp (hunk 3 above)
Comment (abinayaddhandapani) [reply inc/vr_update.hpp:117]: This error is reported as part of moving to c++23 from c++17

File: CMakeLists.txt (hunk 1 above)
Comment (abinayaddhandapani) [reply CMakeLists.txt:4]: Ack

File: inc/vr_update.hpp (hunk 3 above)
Comment (ojayanth) [reply inc/vr_update.hpp:117]: Looks like it is bug . better to fix using different commit. use version change specific commits in this.

**Review by mahkurap — APPROVED** [review]

**Review by supven01 — APPROVED** [review]

**Pull request merged**